internal/compiler/emitter.go

```diff
@@ -20,15 +20,16 @@ const (
 )
 
 type emitter struct {
-	host               EmitHost
-	emitOnly           emitOnly
-	emittedFilesList   []string
-	emitterDiagnostics ast.DiagnosticsCollection
-	emitSkipped        bool
-	sourceMapDataList  []*sourceMapEmitResult
-	writer             printer.EmitTextWriter
-	paths              *outputPaths
-	sourceFile         *ast.SourceFile
+	host                EmitHost
+	emitOnly            emitOnly
+	emittedFilesList    []string
+	emitterDiagnostics  ast.DiagnosticsCollection
+	emitSkipped         bool
+	sourceMapDataList   []*sourceMapEmitResult
+	writer              printer.EmitTextWriter
+	paths               *outputPaths
+	sourceFile          *ast.SourceFile
+	sourcefileMetaDatas map[string]*ast.SourceFileMetaData
```

Comment:
EmitHost already wraps program; I think that the metadatas could just come through that.